application/app/src/main/java/com/example/buber/Views/Activities/LoginActivity.java

```diff
@@ -32,19 +45,56 @@ protected void onCreate(Bundle savedInstanceState) {
 
         App.getModel().addObserver(this);
 
-        usernameEditText = findViewById(R.id.loginUsernameEditText);
-        passwordEditText = findViewById(R.id.loginPasswordEditText);
-        driverLoginButtonId = R.id.loginDriverButton;
+        editEmail = findViewById(R.id.loginEmailEditText);
+        editPassword = findViewById(R.id.loginPasswordEditText);
+        driverLoginBtnID = R.id.loginDriverButton;
+
+        mFirebaseAuth = FirebaseAuth.getInstance();
+        mAuthStateListener = new FirebaseAuth.AuthStateListener() {
+            @Override
+            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+                FirebaseUser mFirebaseUser = mFirebaseAuth.getCurrentUser();
+                if (mFirebaseUser != null) {
+                    Toast.makeText(LoginActivity.this, "You are STILL logged in", Toast.LENGTH_SHORT).show();
+                    // move to the next intent
+                } else {
+                    Toast.makeText(LoginActivity.this, "Please Login", Toast.LENGTH_SHORT).show();
+                }
+            }
+        };
     }
 
     public void handleLoginClick(View view) {
-        ApplicationController c = App.getController();
-
-        String username = usernameEditText.getText().toString();
-        String password = passwordEditText.getText().toString();
-        User.TYPE loginType = view.getId() == driverLoginButtonId ?  User.TYPE.Drivers : User.TYPE.Riders;
-
-        c.login(username, password, loginType);
+        if (LoginFormUtils.validateForm(editEmail, editPassword)) {
+            String email = editEmail.getText().toString();
+            String password = editPassword.getText().toString();
+            User.TYPE loginType = view.getId() == driverLoginBtnID ? User.TYPE.Drivers : User.TYPE.Riders;
+
+            mFirebaseAuth
+                    .signInWithEmailAndPassword(email, password)
+                    .addOnCompleteListener(LoginActivity.this, new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            if (!task.isSuccessful()) {
+                                try {
+                                    throw task.getException();
```

Comment:
why are you just throwing an exception in the try without doing anything else? This seems redundant?